landoapi/uplift.py

```diff
@@ -213,10 +221,76 @@ def create_approval_request(phab: PhabricatorClient, revision: dict, form_conten
     }
 
 
-def stack_uplift_form_submitted(stack_data) -> bool:
+def stack_uplift_form_submitted(stack_data: RevisionData) -> bool:
     """Return `True` if the stack has a valid uplift request form submitted."""
     # NOTE: this just checks that any of the revisions in the stack have the uplift form
     # submitted.
     return any(
         get_uplift_request_form(revision) for revision in stack_data.revisions.values()
     )
+
+
+def create_uplift_bug_update_payload(
+    bug: dict, repo_name: str, milestone: int
+) -> dict[str, Any]:
+    """Create a payload for updating a bug using the BMO REST API."""
+    payload: dict[str, Any] = {
+        "ids": [bug["id"]],
+    }
+
+    milestone_tracking_flag = f"cf_status_firefox{milestone}"
+    if "leave-open" not in bug["keywords"] and milestone_tracking_flag in bug:
+        # Set the status of a bug to fixed if the fix was uplifted to a branch
+        # and the "leave-open" keyword is not set.
+        payload[milestone_tracking_flag] = "fixed"
+
+    checkin_needed_flag = f"[checkin-needed-{repo_name}]"
+    if checkin_needed_flag in bug["whiteboard"]:
+        # Remove "[checkin-needed-beta]" etc. texts from the whiteboard.
+        payload["whiteboard"] = bug["whiteboard"].replace(checkin_needed_flag, "")
+
+    return payload
+
+
+UPDATE_RETRIES = 3
+
+
+def update_bugs_for_uplift(
+    bmo_url: str,
+    bmo_api_key: str,
+    changeset_titles: list[str],
+    repo_name: str,
+    milestone: int,
+):
+    """Update Bugzilla bugs for uplift."""
+    headers = {"X-Bugzilla-API-Key": bmo_api_key}
+    bugs = [str(bug) for title in changeset_titles for bug in parse_bugs(title)]
+    params = {
+        "ids": ",".join(bugs),
+    }
+    bug_endpoint = f"{bmo_url}/rest/bug"
+    resp_get = requests.get(bug_endpoint, headers=headers, params=params)
```

Comment:
Ideally we wrap these calls in a thin client -- something like `BMO.get` so that we only pass the endpoint and params here, and so that this is easier to reuse in the future. With that we reduce the clutter in this method (e.g. the bmo token/url/etc.).